contracts/common/Registry.sol

```diff
@@ -123,6 +136,13 @@ contract Registry is Ownable {
     return isERC721[_token];
   }
 
+  function isTokenMappedAndGetPredicate(address _token) public view returns (address) {
+    if (isTokenMappedAndIsErc721(_token)) {
+      return erc721Predicate;
```

Comment:
somehow I don't like this method. let's say if some erc20/erc721 has multiple predicates?

Reply:
Then we will return the current 20/721 predicate specifically. This predicate needs to be mentioned in the exit to process challenges, if any.